### PR TITLE
Issue19/migrate-to-vite

### PR DESCRIPTION
**Pull request by prak112**
- Manual migration to Vite, Automated migration through cra-to-vite was outdated
- Reorganise codebase :
 - Setup directories
 - Setup frontend service
 - Setup documentation
 - Setup and Refactor components
 - Renamed and setup routing of components
 - Commented possible improvements in components wherever required 
 - Suggested tasks to be implemented in /contexts/UserContext.jsx

### CODE REVIEW

File: src/pages/login-page/LoginPage.jsx
Comment (prak112) [inline src/pages/login-page/LoginPage.jsx:1]: Do you think we could move this to the newly organised Login component ? The codebase is set as perfectly as I did for my own projects.

Comment (prakgov) [inline src/pages/login-page/LoginPage.jsx:1]: Do you think we could add it to the Login component in the reorganised codebase ?

**Review by prak112 — COMMENTED** [review]
- Merge conflicts reasons : 
 - LoginPage has been modified to Login,
 - Overlooked tests directory in previous dir while setting up Vite frontend

File: src/pages/login-page/LoginPage.jsx
Comment (haxwell) [reply src/pages/login-page/LoginPage.jsx:1]: I'm thinking yes.. What's the alternative?

File: package.json
Comment (haxwell) [inline package.json:1]: You will need a package.json somewhere, so as to install dependencies. For instance, see this error:

```
dmpj-folks/prakirth/savvato-work-app-frontend-app$ npm run dev
npm error code ENOENT
npm error syscall open
npm error path /home/jjames/src/dmpj-folks/prakirth/savvato-work-app-frontend-app/package.json
npm error errno -2


npm error enoent Could not read package.json: Error: ENOENT: no such file or directory, open '/home/jjames/src/dmpj-folks/prakirth/savvato-work-app-frontend-app/package.json'

npm error enoent This is related to npm not being able to find a file.


npm error enoent
npm error A complete log of this run can be found in: /home/jjames/.npm/_logs/2024-08-26T03_54_06_544Z-debug-0.log
jjames@hppav06:~/src/dmpj-folks/prakirth/savvato-work-app-frontend-app$ ls
docs  LICENSE  package-lock.json  README.md  work-app-frontend
jjames@hppav06:~/src/dmpj-folks/prakirth/savvato-work-app-frontend-app$ 
```

**Review by haxwell — CHANGES_REQUESTED** [review]

**Comment by haxwell** [comment]
FYI, I'll continue reviewing once the package.json issue is addressed.

File: package.json
Comment (prakgov) [reply package.json:1]: It exists inside the directory, work-app-frontend/. I assume, you happen to run the project from the main directory.

Comment (haxwell) [reply package.json:1]: Got it. I see it now.

**Comment by haxwell** [comment]
Okay. I've got it running now. Using Vite is working out. We can keep that. But the UI changes, those need to be reverted. 

We do have an issue (#16 ) to set the default pages (so that you first see the login page, with a create user option, and then once logged in see the home page. From there, you could create a step). So that will be addressed soon.

This PR should be about the plumbing, the infrastucture, but nothing the user sees should change in this PR. See #16

**Review by haxwell — CHANGES_REQUESTED** [review]
Moving the files, setting up a directory structure, etc, is all fine. But there is no need to change what the user sees in this PR. Revert any UI changes, other than relocating the files.

**Comment by prak112** [comment]
- I agree and understand that this was out of the context of the issue. But I would like you to consider the possibility as described below.
- The [user will still be redirected to registration](https://github.com/prak112/savvato-work-app-frontend-app/blob/issue19/migrate-to-vite/work-app-frontend/src/components/Home.jsx) for authentication (this was just for display). 
``` javascript

// work-app-frontend/src/components/Home.jsx
....

    /**
     * Redirects to the authentication page.
     */
    // const verifyUser = () => {
    //     // if authorized navigate to steps
    //     // if not, navigate to signup and then login
    // }

....

```
- This redirection logic needs to be implemented via [UserContext.jsx](https://github.com/prak112/savvato-work-app-frontend-app/tree/issue19/migrate-to-vite/work-app-frontend/src/contexts)
- My motivation to change the UI was to give a sneak-peek to the user on what exactly the application does. Since it is not clear in any way about what the application is. 
- Also, when a user is just asked to register without giving an idea of what they are registering for, would be cruel. 
- Consider this and let me know. I will redo the changes if you are not convinced of my logic.

**Comment by prak112** [comment]
Sorry about that. Still learning to work with PRs and reviewing 😄

**Comment by haxwell** [comment]
The Unix Principle, do one thing and do it well.. is my motivation here! :)

**Comment by haxwell** [comment]
I just realized I did not, at least in this PR, directly respond to your last message.

"My motivation to change the UI was to give a sneak-peek to the user on what exactly the application does. .. [and more good points].."

I'm not convinced :) I would respond that the user is not looking at these changes. They are all valid, and useful, certainly as proof of concept. But they belong in a separate PR. Your audience here is other developers, not the end user. Gotta make the PRs for developers, even when they benefit the end user.

**Pull request merged**